File: 09.Greed贪心/leetcode_1877_minPairSum.py
from typing import List

class Solution:
    def minPairSum(self, nums: List[int]) -> int:
        nums.sort()
        res = 0
        for i in range(len(nums)//2):
            cur = nums[i]+nums[-i-1]
            if cur > res:
                res = cur
        return res

if __name__ == "__main__":
    s = Solution()
    nums = [3,5,2,3]
    print(s.minPairSum(nums))

File: 09.Greed贪心/leetcode_1877_minPairSum.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int minPairSum(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int res = 0;
        int n = nums.size();
        for (int i=0; i<n/2; i++){
            res = max(res, nums[i]+nums[n-i-1]);
        }
        return res;
    }
};

int main(){
    Solution s;
    vector<int> nums = {3,5,2,3};
    cout << s.minPairSum(nums) << endl;
}
